src/ngraph/runtime/intelgpu/intelgpu_op_custom_kernels.cpp

```diff
@@ -92,6 +92,29 @@ string
     return buffer;
 }
 
+void runtime::intelgpu::generate_function_definition(codegen::CodeWriter& writer,
```

Comment:
I would propose to change the interface.
It might be better (discussable) to use following:
```
gen_func_def(
codegen::CodeWriter& writer,
const string& entry_point_name,
const vector<string>& input_types, 
const vector<string>& input_shapes, 
const string& output_type, 
const string& output_shape, 
)
```
Assume all inputs are started with `"input"` string and a number (input0, input1, etc.) added in a loop.